pkg/controllers/deliverable_reconciler.go

```diff
@@ -169,17 +170,20 @@ func (r *DeliverableReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 	}
 
-	return r.completeReconciliation(ctx, deliverable, realizedResources, err)
+	return r.completeReconciliation(ctx, deliverable, resourceStatuses, err)
 }
 
-func (r *DeliverableReconciler) completeReconciliation(ctx context.Context, deliverable *v1alpha1.Deliverable, realizedResources []v1alpha1.RealizedResource, err error) (ctrl.Result, error) {
+func (r *DeliverableReconciler) completeReconciliation(ctx context.Context, deliverable *v1alpha1.Deliverable, resourceStatuses statuses.ResourceStatuses, err error) (ctrl.Result, error) {
 	log := logr.FromContextOrDiscard(ctx)
 	var changed bool
 	deliverable.Status.Conditions, changed = r.conditionManager.Finalize()
 
 	var updateErr error
```

Comment:
nit: seems like `updateErr` is only used in place inside the branch below - I think we could go with a straight `:=` right there instead